### DISCUSSION TITLE
Who Pushes the Push Notifications?

### DISCUSSION THREAD

**Discussion by LinuxinaBit** [berty/berty#4601] [Q&A]
If Berty is P2P, and all data is stored on device, who can decrypt your messages and send a push notification?

**Comment by LFGaming** [comment]
You and the contacts you are chatting with are the only once that can decrypt your messages. See: [the whitepaper, encryption](https://berty.tech/docs/protocol/#encryption) or [the whitepaper itself](https://berty.tech/docs/protocol/). As far as I know nothing from the notification system can read the message, but not sure how that works.
